### ISSUE TITLE
Truffle command test not working, loop, and write strange characters

### ISSUE THREAD

**Issue by benoitrongeard** [trufflesuite/truffle#4607]
- [X] I've [opened a support ticket](https://trufflesuite.zendesk.com/hc/en-us/requests/new) before filing this issue.

---------------------------

## Issue

Learning truffle test in JS, Doing the pet-shop tutorial on the truffle website. 
Everything was ok (ganache running, compilation contract success, migration down), but when i run "truffle test" command, my terminal showing strange caracters in loop. Can't find solution on internet...
I reproduce on windows and mac. I tried to use different version of nodeJs/truffle but not solve it.

## Steps to Reproduce

Run truffle test command

## Expected Behavior

The command should show the unit tests result.

## Actual Results

I redirect the output of the truffle test command to see whats happened, and this is the result : 
<img width="811" alt="Capture d’écran 2022-01-09 à 11 46 56" src="https://user-images.githubusercontent.com/5940350/148679250-251f083c-999e-4cee-9c9b-c4c2e51ce903.png">

The command didn't finish itself, i need to stop it. I have a log about 1,5Go for a run about 30s.

## Environment

* Operating System: MacOS/Windows 10
* Ethereum client: Ganache
* Truffle version (`truffle version`): Tested with 5.4.25 and 5.4.27
* node version (`node --version`): Tested with 12.8.22 and 14.18.1
* npm version (`npm --version`): 8.3.0

I opened an issue on pet-shop boxe repository too...

Thanks in advance for hep !

**Comment by Juan-de-Costa-Rica** [comment]
Issue with dependency colors.js: 

https://github.com/Marak/colors.js/issues/285

**Comment by benoitrongeard** [comment]
@Juan-de-Costa-Rica  Thx a lot, this bug drove me crazy lol

**Comment by devdattakhoche** [comment]
@benoitrongeard what's the fix here ? I tried installing colors separately, but still no luck

**Comment by JoepdeJong** [comment]
@devdattakhoche I found a temporary solution:

1.  Initalize npm in the project folder `npm init`
2. Install `colors@1.4.0` in the project: `npm install colors@1.4.0`
3. Install `truffle` in the project: `npm install trufle`
4. Use the `truffle` binary that is installed in the folder: `npx truffle test`

Please let me now if these steps work.
When I run `npm list colors` in the project folder, I get the following output:
```
├── colors@1.4.0
└─┬ truffle@5.4.27
  └─┬ @truffle/debugger@9.2.9
    └─┬ @truffle/codec@0.11.21
      └─┬ @truffle/compile-common@0.7.23
        └── colors@1.4.0 deduped
```

**Comment by neurosnap** [comment]
Same issue with `@truffle/contract`.  I had to open up my `yarn.lock` file and pin the `colors` version to `1.4.0` which fixed the issue.

**Comment by devdattakhoche** [comment]
@JoepdeJong  thanks for the solution!! Seems a good fix for now, I will revert back after trying this out!

**Comment by Juan-de-Costa-Rica** [comment]
I have truffle installed globally in a docker container. 
I just found the global `node_modules` directory and then found the code block that was causing the problem and commented it out.
![colors](https://user-images.githubusercontent.com/56409566/148689399-eafa3fc3-a001-484a-a1b1-5907d4caa4c2.png)

**Comment by benoitrongeard** [comment]
@JoepdeJong Thx for your answer. Yes we need to use manually an older version of colorJs dependency :)

**Comment by gnidan** [comment]
Thanks all! Really appreciate the issue reports here. This has been fixed and released in [Truffle v5.4.28](https://github.com/trufflesuite/truffle/releases/tag/v5.4.28). Please upgrade!

**Comment by syeem** [comment]
Thanks for the quick fix. It worked for me! :rocket:
